modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -18,111 +18,142 @@ InputParameters validParams<MultiSmoothSuperellipsoidIC>()
+  _bv = _semiaxis_b_vec.size();
+  _cv = _semiaxis_c_vec.size();
 
-  //Set up domain bounds with mesh tools
-  for (unsigned int i = 0; i < LIBMESH_DIM; i++)
+  if (_nv != _bsv || _nv != _ev || _nv != _av || _nv != _bv || _nv != _cv)
+    mooseError("Vectors for numbub, bubspac, exponent, semiaxis_a, semiaxis_b, and semiaxis_c must be the same size.");
+
+  if (_semiaxis_variation_type != 2)
   {
-    _bottom_left(i) = _mesh.getMinInDimension(i);
-    _top_right(i) = _mesh.getMaxInDimension(i);
+    _avv = _semiaxis_a_variation_vec.size();
+    _bvv = _semiaxis_b_variation_vec.size();
+    _cvv = _semiaxis_c_variation_vec.size();
+
+    if (_nv != _avv || _nv != _bvv || _nv != _cvv)
+      mooseError("Vectors for numbub, semiaxis_a_variation, semiaxis_b_variation, and semiaxis_c_variation must be the same size.");
   }
-  _range = _top_right - _bottom_left;
 
-  if (_semiaxis_a_variation > 0.0 && _semiaxis_variation_type == 2)
-    mooseError("If Semiaxis_a_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
-  if (_semiaxis_b_variation > 0.0 && _semiaxis_variation_type == 2)
-    mooseError("If Semiaxis_b_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
-  if (_semiaxis_c_variation > 0.0 && _semiaxis_variation_type == 2)
-    mooseError("If Semiaxis_c_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
+  for (unsigned int k = 0; k < _nv; k++)
+  {  //Set up domain bounds with mesh tools
+    _numbub = _numbub_vec.at(k);
+    _bubspac = _bubspac_vec.at(k);
+    _exponent = _exponent_vec.at(k);
+    _semiaxis_a = _semiaxis_a_vec.at(k);
+    _semiaxis_b = _semiaxis_b_vec.at(k);
+    _semiaxis_c = _semiaxis_c_vec.at(k);
+    _semiaxis_a_variation = _semiaxis_a_variation_vec.at(k);
+    _semiaxis_b_variation = _semiaxis_b_variation_vec.at(k);
+    _semiaxis_c_variation = _semiaxis_c_variation_vec.at(k);
+
+    for (unsigned int i = 0; i < LIBMESH_DIM; i++)
+    {
+      _bottom_left(i) = _mesh.getMinInDimension(i);
+      _top_right(i) = _mesh.getMaxInDimension(i);
+    }
+    _range = _top_right - _bottom_left;
+
+    if (_semiaxis_a_variation > 0.0 && _semiaxis_variation_type == 2)
+      mooseError("If Semiaxis_a_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
+    if (_semiaxis_b_variation > 0.0 && _semiaxis_variation_type == 2)
+      mooseError("If Semiaxis_b_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
+    if (_semiaxis_c_variation > 0.0 && _semiaxis_variation_type == 2)
+      mooseError("If Semiaxis_c_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
 
-  SmoothSuperellipsoidBaseIC::initialSetup();
+    SmoothSuperellipsoidBaseIC::initialSetup();
+  }
 }
 
 void
 MultiSmoothSuperellipsoidIC::computeSuperellipsoidSemiaxes()
 {
-  _as.resize(_numbub);
```

Comment:
You've changed the pre-existing code from just filling a vector to using `push_back()`.  Was this a conscious decision?  Why not at least reserve `_numbub` entries?

I'm not arguing that it's an efficiency issue since it's `initialSetup()` code, but I also don't see a reason to change it if it was working before.